chart/values.yaml

```diff
@@ -587,6 +587,34 @@ loki-stack:
               - __meta_kubernetes_pod_uid
               - __meta_kubernetes_pod_container_name
               target_label: __path__
+
+nats:
+  enabled: true
+  nats:
+    image:
+      repository: nats
+      pullPolicy: IfNotPresent
+    jetstream:
+      enabled: true
+      memStorage:
+        enabled: true
+        # Size of nats message is around 0.3 KB, so it can store around 10K messages.
+        size: "5Mi"
+      fileStorage:
+        enabled: false
+  cluster:
+    enabled: true
+    # Cluster name is required, by default will be release name.
+    name: "nats"
+    replicas: 3
+  # The nats box can be installed for debugging
```

Comment:
Why are we using clustering?

Reply:
Probably because we're using in-memory, so if one pod restarts, the other ones still keep the messages? @datacore-vvarakantham

Reply:
Should be have 2 factor redundancy here? We're using the CR/configmap for persistent storage on the cluster etcd, and it's not mission critical..

Reply:
Is the 'name' key used? I don't see the option here on the latest version of the 'nats' helm chart. Refer to: https://github.com/nats-io/k8s/blob/v0.13.1/helm/charts/nats/values.yaml#L336

Reply:
Yes. In cluster mode the servers connect to each other. And the cluster self heals if any of the server is down.

Also this is how its described in Nats docs for stream replication.

Replicas=1 - Cannot operate during an outage of the server servicing the stream. Highly performant.
Replicas=2 - No significant benefit at this time. We recommend using Replicas=3 instead.
Replicas=3 - Can tolerate loss of one server servicing the stream. An ideal balance between risk and performance.